.envrc

```diff
@@ -0,0 +1,153 @@
+    openssl         ;
+    do
+    if ! require "$pkg"; then
+        die "You seem to be missing the system dependency: ${pkg}
+Please install homebrew, and run brew bundle."
+    fi
+done
+
+### Python ###
+
+# direnv set -u's; unsure if it's a good idea to set +u.
+VIRTUAL_ENV="${VIRTUAL_ENV-}"
+
+if [ -n "$VIRTUAL_ENV" ]; then
+    # we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
+    if [ "$VIRTUAL_ENV" != "${PWD}/.venv" ]; then
+        info "You're in a virtualenv, but it's not in the expected location (${PWD}/.venv)"
+        init_venv
+    fi
+else
+    if [ ! -f ".venv/bin/activate" ]; then
+        info "You don't seem to have a virtualenv."
+        init_venv
+    fi
+fi
+
+info "Activating virtualenv."
+source .venv/bin/activate
+[ "$(command -v python)" != "${PWD}/.venv/bin/python" ] && die "Failed to activate virtualenv."
+
+python -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))" || \
+    die "For some reason, the virtualenv isn't Python 2.7."
+
+# naive check, but we need to stay fast
+if [ "$(command -v sentry)" != "${PWD}/.venv/bin/sentry" ]; then
+    info "Your .venv is activated, but sentry doesn't seem to be installed. Let's install it."
+    make ensure-pinned-pip
+    # TODO: use venv-update to create venv. this assumes virtualenv is installed
+    # which is reasonable if python2.7 is via pyenv. I forget if brew python@2 contains it.
+    SENTRY_LIGHT_BUILD=1 make install-sentry-dev
+fi
+
+### pre-commit ###
+
+# this is cheap, so we'll just do it every time
+ln -sf config/hooks/* .git/hooks
+
+if ! require pre-commit; then
+    info "Looks like you don't have pre-commit installed. Let's install it."
+    make setup-git
+fi
+
+# this hotfix is cheap too, so just run it every time
+rm -f .git/hooks/pre-commit.legacy
+
+### devservices/database ###
+
+SENTRY_DIRENV_NO_CHECK_DEVSERVICES="${SENTRY_DIRENV_NO_CHECK_DEVSERVICES-}"
+if [ -z "${SENTRY_DIRENV_NO_CHECK_DEVSERVICES}" ]; then
+    info "Checking for devservices containers, this might be a bit slow.
+To disable in the future, please set SENTRY_DIRENV_NO_CHECK_DEVSERVICES=1"
+
+    # XXX: these container names are hardcoded for now
+    # NOTE: sentry_symbolicator isn't started up by devservices, and is behind a config flag, so we're not checking for it
+    for container in \
+        sentry_postgres     \
+        sentry_clickhouse   \
+        sentry_snuba        \
+        sentry_redis        ;
+        do
+        docker exec "$container" true || die "The docker container ${container} doesn't seem to be running.
+    Please run sentry devservices up."
+    done
+fi
+
+# TODO: is there a faster way to state-check the need for sentry reset-db and sentry createuser?
+# This is a little slower than i'd like, so not turning it on for now, not even behind a NO_CHECK flag.
+# docker exec sentry_postgres sh -c "psql -U postgres -h 127.0.0.1 sentry --command 'select * from sentry_useremail'"
```

Comment:
Slightly faster query would just just be `select 1 from sentry_useremail`. I'm not sure if that's the source of it being slow or not though.

Reply:
With my dev database, there's only 1 user - speed is roughly the same haha. Thanks though.